9주차 미션 / 안드로이드 2조 김태정

imtaejugkim opened the pull request.

imtaejugkim committed:
test commit

imtaejugkim committed:
1주차 커밋 수정

nonaninona commented:
미션 제출물에 대해서 개선 해볼만한 사항 말씀드릴게요~

이미 잘하시니까 디테일을 챙기셔서 xml을 예쁘게 작성해보시면 좋을 것 같습니다.
후에 프로젝트 진행하면서 디자이너의 화면을 받아서 그대로 적용하는데에 큰 도움이 될 것 같아요.

예를 들면,

- 뷰들의 간격을 조정하는 것과
- 폰트의 굵기, 색, 크기를 설정하는 것
- 이미지의 크기, 색상(tint)를 설정하는 것
- 구분선의 색상, 간격을 설정하는 것
- status bar 색상 설정하는 것

이 있겠네요

xml 코드 상에서 구조적으로 아쉬운 점은 제 수준에선 없었고 ^~^
스터디 때 말씀하신 맨 위의 메뉴바에 대해서

- transprent 설정
- 스크롤 내리면서 색 변화
- 맨 위 고정

을 도전해보시는 것도 좋을 것 같습니다!

imtaejugkim committed:
상위 풀더 수정

imtaejugkim commented:
루트 변경 완료했습니다! 혹시 오류가 발생한다면, 말씀해주시면 감사하겠습니다!

nonaninona commented:
루트 변경된 프로젝트 잘 돌아가는 것까지 확인했습니다!

imtaejugkim committed:
2주차 커밋

imtaejugkim commented:
2주차 미션 업로드 완료했습니다!

imtaejugkim committed:
2주차 commit 추가

- 알림 화면 수정
- statusbar scroll 시 색깔 변경

imtaejugkim commented:
1주차 scroll시 색상 변경 / 2주차 알람 화면 변경 완료했습니다!

imtaejugkim committed:
Update AlarmActivity.kt

imtaejugkim commented on app/src/main/java/com/example/carrotmarket/AlarmActivity.kt:
```diff
@@ -1,6 +1,7 @@
 package com.example.carrotmarket
 
 import android.opengl.Visibility
+import android.os.Build.VERSION_CODES.R
```
이 import 문을 삭제해야 unresolved reference 오류가 발생하지 않는 것 같습니다..!

nonaninona replied:
확인했습니다~ 아마 Resource 객체인 R과 겹쳐서 그런 것 같네용

nonaninona commented on app/src/main/AndroidManifest.xml:
```diff
@@ -13,13 +13,21 @@
+        <activity
+            android:name=".StuffInfoActivity"
+            android:exported="false"
+            android:noHistory="true" />
```
activity stack에 쌓지 않는 옵션인데 추가하신 이유가 있을까요?

imtaejugkim replied:
다른 activity로 옮겨질 때, back stack에 저장하지 않아야 한다고 생각해서 추가했습니다.
다시 생각해보니, 회원가입 화면과 같이 일회성 작업이 아닌데 불필요하게 추가 한 것 같습니다!
stack 관리를 조금 더 생각해보고 작성하도록 하겠습니다!

nonaninona commented on app/src/main/res/layout/activity_alarm.xml:
```diff
@@ -0,0 +1,132 @@
+            android:text="알림"
+            android:textSize="15sp"
+            android:textStyle="bold"
+            android:layout_marginLeft="20dp"
```
marginLeft보단 marginStart가 좋습니다~
https://latte-is-horse.tistory.com/221

imtaejugkim replied:
marginStart로 사용하겠습니다!

nonaninona commented on app/src/main/res/layout/activity_alarm.xml:
```diff
@@ -0,0 +1,132 @@
+            android:textSize="15sp"
+            android:textStyle="bold"
+            android:layout_marginLeft="20dp"
+            android:src="@drawable/tab_icon_home_n"
```
???

nonaninona commented on app/src/main/res/layout/activity_alarm.xml:
```diff
@@ -0,0 +1,132 @@
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/cl_top_appbar"
+        android:orientation="horizontal"
+        android:layout_marginTop="10dp">
```
LinearLayout의 layout_weight 속성을 사용하지 않고
constraintLayout만으로도 구현할 수 있습니다!

hint : width 0dp

nonaninona commented on app/src/main/res/layout/activity_stuffinfo.xml:
```diff
@@ -0,0 +1,228 @@
+                        <LinearLayout
+                            android:layout_width="wrap_content"
+                            android:layout_height="wrap_content"
+                            android:layout_marginRight="10dp"
```
left, start와 같은 이유로 end가 좋습니다~

nonaninona commented on app/src/main/res/layout/activity_stuffinfo.xml:
```diff
@@ -0,0 +1,228 @@
+                            <TextView
+                                android:layout_width="wrap_content"
+                                android:layout_height="wrap_content"
+                                android:textSize="18dp"
```
sp와 dp의 차이를 아신다면 의도적인 것이지만 만약 그렇지 않다면
일반적으로 sp를 권장합니다~

모르신다면 이참에 알아보아요
https://woovictory.github.io/2019/01/03/Android-Diff-of-DP-SP/

imtaejugkim replied:
앗 자동완성을 쓰다가.. 오타가 난 것 같습니다. 감사합니다!

nonaninona reviewed (changes requested):
2주차 미션 코드 리뷰 내용입니다~

오 와우 알람 화면에서의 탭 전환도 구현하셨네요!
5주차에 배우는 탭 레이아웃에서 수정하게 될테니 기대해주세용~

역쉬 기본기 있으신게 보입니다!
2조는 다들 너무 잘해서 저말고 현석이형님께 부탁드려야겠어요 ㅜㅜ

imtaejugkim committed:
3주차 커밋

1. my 페이지 레이아웃 구현
2. 메인 화면 글쓰기 버튼 구현
3. 메인 상세 화면 하단 bar 고정

imtaejugkim committed:
2주차 피드백 수정

1. 오타 수정 (dp->sp)
2. layout_weight 대신 0dp 활용 -> guideline을 50퍼센트로 부여하여, 0dp 및 제약조건을 통해 설정
3. manifest back stack 관리 삭제
4. marginLeft대신 marginStart로 변경

imtaejugkim committed:
4주차 커밋

1. RecyclerView 작성
2. Adapter 작성
3. intent를 통해 데이터 클래스 전달
4. 상세 페이지마다 적용

imtaejugkim commented:
4주차 미션 업로드 했습니다!

imtaejugkim committed:
5주차 커밋

1. 알림화면 활동알림 / 키워드 알림 tabLayout과 Viewpager2로 화면 전환
2. 각 알림 recyclerview 구현
3. 웹에서 Glide를 사용하여 가져온 image를 imageList에 추가하여, viewpager2로 전환하며 출력

imtaejugkim commented:
5주차 미션 업로드 했습니다!

imtaejugkim committed:
6주차 커밋

1. thread() 를 이용해 sleep 시간 차이로 화면을 자동으로 스와이프

imtaejugkim commented:
6주차 미션 업로드 했습니다!

imtaejugkim committed:
7주차 커밋

Room DB, DAO, Entity를 통해 데이터베이스 구축

imtaejugkim commented:
7주차 미션 업로드 했습니다!

imtaejugkim commented:
7주차 미션 업로드 했습니다!

imtaejugkim committed:
9주차 커밋

9주차 기본 미션만 완료하고 추가 찜하기 기능은 하지 못했습니다.
다음주까지 기능 구현 한 번 시도해보겠습니다!

imtaejugkim commented:
9주차 미션 업로드 했습니다!
